### ISSUE TITLE
homepage showing some json file content from jetpack sync

### ISSUE THREAD

**Issue by geekz** [Automattic/jetpack#20598]
#### Steps to reproduce the issue
The site has object cache via redis and page cache via nginx is implemented.  It also has jetpack enabled ofcourse. 
The site suddenly shows json code as given below in homepage. I belive the json code is from jetpack sync. 

#### Server configuration
The server has openresty server software installed. and it whitelists only the following query parameters :  "amp", "p", "relatedposts".

#### What happened instead
 Sometimes suddenly it shows json code as given below.
`{"objects":"rVfbjuM2DP2Vwq9N0rkVg52nPrQP\/Ya2MGRJdjSj20qy45n1\/ntJOYllWw4Wg8XA4\/DwkKJIUZa+Fb8\/filevhWMBILvv\/8sXiK2K1rPXSlNIzRAQtfmj5OtzLs\/UKMuai0o10TxM2Mk7BMCV0TIbfvWofIiOd4IH7jjDMCHu\/sv+7vn\/f3jLw\/3L4\/PL\/ePFx6hQXQkCKPLN\/6eOPCBhNYDcAcIE95K8l4m8c0jqCtqtOY0lJL4cJnp3UyDbgU7w0RKc+KsVELxMrxb7jFlr7YZXi2P\/3AcRRr+72+IgEkj6glDYVdY3UwQCruiUnaCUNgVQdT1EFJrREBx4lVCjtKuENRMWL8HUQN65IJOcJRgEr4eiO8B7wTjBunK7wFF36pb4IggvuQjgvgKxhE6McEoYC26xEOUIMtyNhqKu0KZbvgaJvxrK+hbgISjDrOsLD58Yqgx08o+DeqpS\/EngE2TQKZpxgwmcUcJzN\/mUyfBGf9GQPPY2AEeO6mjhIoHVDzMFQ9YvD5AjukAf3Q4Dt7hhALvAxRcEoG1ob67YvgbjBIkkGrvuSWOBM72HZEtrDUss5+MiOSaEQewC\/0VdoIe8VccImF7tD8GNcAjr3AUdkUXpgDxNxSs7nHCxFopaGw1CCqQ+tc+Gij7CMkm8FTIapmYCkEInTAUIEAyOKImtN87TmQUsB6km1QoYNWawTRkgsfK1TL1HSWIRbABHpEEghKuTpIOGZcnieP1S5z0cQnM+dMScKFeJgOhXfHql\/gr6aDyTljMomUrO4SgNLDjZE1xHUBNF5p+H2KhP8SqKAjBNvMxNBllv29GvVv7dNHj88fa5vkDd3DruPewE8MsPFuSDA087H2AMmLz9HQ1zwWDGbpkKH8yLvo3YbDWwxOWnE6zA9THwrbrrBEak3pyYoMGmoD7RC\/J0EsPT4DntMHmPeVxLbNqg0Eo5bFvlF2l9hKZM6\/wnRhn2OdIxnINTVMbBwvK701dwzcTyK3iOhwwBeADxxG6UfJw0Ywe1dZMwexKPShCnflLk0pydrh\/iKbhZwQTuIL9KuYUPN4M5kLNBAOl+EQwHhYgYf7IOexRh\/gefW2FEes5EvNBbJU5NayEJu49a\/+ZjM4nkaQT3N2ex818ktu2hDGhM4bQYJ+YA24E8I58mEQqjj63gpm6dmaUDewzK2QRmJfwEfZHcxod\/kBUV4tcSJ\/qoEVIScHtdv8kEd2ourWbVU8cbJXeS\/aTMjw6+9HsZkIxmgdDh\/H1EN\/K4su+NauviObaxPyZ3q4+mBGDflhrztGAJtlPDct+XwzxwsdEXOd9PkAZlt32MwaLpjAsG1HGMNkgot0qAxt2jSP2iMdBNFp9WjeM6JG4cZjsVpixwGthRXwswPoss2GEi6mV8aRlh5NdHR7iZ4a7evxujje4lV8E+AGU5\/rrVlXcZdM6Ui8E6BS47twgjurvu6Is4dJ0Dz\/w0gt3XjiWERuvdIQpoeEqCsdvAwen4Fr+PWrP982TLUEglZAiiHgyd0bimP8sTP+Ll8aLW38SgR7LcOQKyeh2V3AmwgiVxmKsV43QcJ+VcsE\/X355aWULN9YVewFH93iLvSKKaMjAcjBlGMfbRglHP6zikg555I1xgi8VUui3K9ZaaQgrayEnnlDW4AXozNCgjHf8cryJJGFa48M8cANzdz6jsG0lBbQNm+vOcM5gXBQjgj23MkkJs8FXimTwVMe45FiXDJZzdaHfdpZOZOEso9pIy0XrzitnnZ2cBrOU1yz9rTOU08z9ZSY7W6YU6Ess+mbEHytD3LWCtPXBKPHBp1XIMg1yCXqOnrnzHjuD1LirT0jroo8cV6ZbBAif3zkAx2llltM4R7LYB\/pZm4yNtAo2oqkh7EtjQxUvupXyuqt9\/x8=","codec":"deflate-json-array"}`

**Comment by jeherve** [comment]
That's an odd one. This is indeed generated by Sync, but I do not understand why it would ever get displayed on the frontend of your site.

Did this start happening when you updated to the most recent version of Jetpack, 10.0, that was released last week, or did this start happening after another change on your site or server?

**Comment by geekz** [comment]
No. I recently updated it to Jetpack 10.0. It was happening before updating as well.  I did one change. I whitelisted query parameter "sync" as well. and now i think i no longer have that problem.

**Comment by jeherve** [comment]
That is a bit odd, those are responses from API endpoints, so that's not data that is available by just adding a query parameter to a URL that's available on the frontend.

Do you have the option to whitelist API requests (everything starting with `yourdomain.com.wp-json/`)?
